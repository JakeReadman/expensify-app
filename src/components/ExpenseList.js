import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from '../components/ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
  <div className='content-container list-body'>
    {props.expenses.length === 0 ? (
      <div className='list-header'>No Expenses to Show</div>
    ) : (
      <div className='list-header'>
        <div className='show-for-mobile'>Expenses</div>
        <div className='show-for-desktop'>Expense</div>
        <div className='show-for-desktop'>Amount</div>
      </div>
    )}
    {props.expenses.map((expense) => (
      <ExpenseListItem key={expense.id} {...expense} />
    ))}
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters),
  };
};

export default connect(mapStateToProps)(ExpenseList);
