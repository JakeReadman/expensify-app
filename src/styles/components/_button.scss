.btn {
  background-color: $light-blue;
  border: none;
  border-radius: 3px;
  color: white;
  display: inline-block;
  font-size: $m-size;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
}

.btn:hover {
  opacity: 0.8;
}

.btn--link {
  background: none;
  font-weight: 200;
  letter-spacing: 0.1rem;
}

.btn--secondary {
  background-color: #888;
}
