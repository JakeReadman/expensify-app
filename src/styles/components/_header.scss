.header {
  background: $blue;
}

.header__content {
  align-items: center;
  display: flex;
  justify-content: space-between;
  padding: $s-size 0;
}

.header__title {
  color: white;
  text-decoration: none;
  h1 {
    margin: 0;
    transition: color 0.3s ease;
  }
}

.header__title h1:hover {
  color: $blue-shadow;
}
