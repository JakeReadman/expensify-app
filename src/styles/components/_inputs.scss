.text-input {
  border: 1px solid #cacccd;
  height: 50px;
  font-size: $m-size;
  font-weight: 300;
  padding: $s-size;
}

.select:hover {
  cursor: pointer;
}

.select {
  @extend .text-input;
}

.date-picker:hover {
  cursor: pointer;
}

.textarea {
  @extend .text-input;
  height: 10rem;
  resize: none;
}
