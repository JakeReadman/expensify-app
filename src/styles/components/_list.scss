.list-header {
  background-color: $off-white;
  border: 1px solid darken($off-white, 7%);
  color: $grey;
  display: flex;
  justify-content: space-between;
  padding: $s-size $m-size;
}

.list-body {
  margin-bottom: $l-size;
}

.list-item {
  border: 1px solid darken($off-white, 7%);
  border-top: none;
  color: $dark-grey;
  display: flex;
  flex-direction: column;
  padding: $s-size;
  text-decoration: none;
  transition: background-color 0.4s ease;
  &:hover {
    background-color: darken($off-white, 4%);
  }
  @media (min-width: $desktop-breakpoint) {
    align-items: center;
    flex-direction: row;
    justify-content: space-between;
    padding: $m-size;
  }
}

.list-item__title {
  margin: 0;
  word-break: break-all;
}

.list-item__sub-title {
  color: $grey;
  font-size: $s-size;
}

.list-item_data {
  margin: $s-size 0 0 0;
  @media (min-width: $desktop-breakpoint) {
    margin: 0;
    padding-left: $s-size;
  }
}
