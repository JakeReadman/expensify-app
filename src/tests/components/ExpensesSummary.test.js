import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('Should render ExpensesSummary component with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});

test('Should render ExpensesSummary component with single expense', () => {
  const wrapper = shallow(<ExpensesSummary expenses={[expenses[1]]} />);
  expect(wrapper).toMatchSnapshot();
});
