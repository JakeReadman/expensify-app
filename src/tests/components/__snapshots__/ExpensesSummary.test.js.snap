// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary component with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span />
       
      expenses
       totalling 
      <span>
        $0.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <[object Object]
        className="btn"
        to="/create"
      >
        Add Expense
      </[object Object]>
    </div>
  </div>
</div>
`;

exports[`Should render ExpensesSummary component with single expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span />
       
      expenses
       totalling 
      <span>
        $0.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <[object Object]
        className="btn"
        to="/create"
      >
        Add Expense
      </[object Object]>
    </div>
  </div>
</div>
`;
